www/source/blog/2017-10-27-distributed-java-application.html.md

```diff
@@ -0,0 +1,360 @@
+---
+title: "Modernize your Java development workflow with Habitat"
+date: 2017-10-27
+author: Matt Wrock
+tags: Java, tutorial, tomcat, mongodb, docker, packaging
+category: community
+classes: body-article
+---
+
+![](media/2017-10-27-modernize-java/azureopendev.jpg)
+
+This week I had the honor of speaking at Microsoft Azure's [Opendev](https://azure.microsoft.com/en-us/opendev/), which hosted a variety of speakers from various cool open source projects and I represented Habitat. I raced through a tour of building a distributed Java web application in Habitat. I showed the application running in a local Habitat Supervisor and also had it perform a rolling update accross three nodes in Azure. Finally I demonstrated the same application running in a container on an Azure deployed Docker host. You can listen to the presentation which starts at about 1:22:20 into the recorded video.
+
+In this post I'd like to cover the same tour in blog format which might allow this information to sink in more effectively for those who better digest written material (like myself) and for the benefit of the google indexing servers. This is very much a beginner level post and I hope that those who have little to no Habitat knowledge will walk away with an understanding of what Habitat is and how it can help step up their development workflow in this age of distributed computing.
+
+## What is Habitat?
+
+As a intro level topic, we are going to start at the very beginning and ask what Habitat is. I have been working on the Habitat code base for a year now and have been working for the company who created Habitat - [Chef](https://www.chef.io/) -  since its begginning and even I still find this a difficult question to answer. I think it can be tough to do the question justice because:
+
+1. There is no other project quite like Habitat. I can't say "Oh Habitat? Its a blang bloom system. Ya know just like jitter bug - that other blang bloom system."
+2. It covers alot of ground and thus I cant simply say "it's a build system" or a "service discovery component" even though it can be both of those things and more.
+3. Certain features like its Docker image exporter can distract people and makes them quickly draw inaccurate conclusions that Habitat is a container runtime.
+
+So here it goes: Habitat provides an environment (a "Habitat") that builds, packages and runs your applications. This environment isolates both the internal (library) and external (service) dependencies of your core application so that packages that run in one environment are guaranteed to run in any other environment. Further, It does not matter what platform the environment runs on. On linux, any modern 64 bit kernel can play. These packages, running inside of Habitat can easily discover and share information about one another making it natural for distributed systems to "bind" together and form your complete application.
```

Comment:
I hadn't heard the Habitat term "bind" used in quite this way before; I like the mental model it encourages of a distributed app binding together.